### ISSUE TITLE
Intro to PyTorch - Part 3 - pb next(iter(trainloader))

### ISSUE THREAD

**Issue by temanava** [udacity/deep-learning-v2-pytorch#154]
Hello, I have a problem with this line of code:
images, labels = next(iter(trainloader))

Here is the error:
RuntimeError                              Traceback (most recent call last)
<ipython-input-23-d77be68f5b26> in <module>()
      8 
      9 criterion = nn.NLLLoss()
---> 10 images, labels = next(iter(trainloader))
     11 images = images.view(images.shape[0], -1)
     12 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\dataloader.py in __next__(self)
    613         if self.num_workers == 0:  # same-process loading
    614             indices = next(self.sample_iter)  # may raise StopIteration
--> 615             batch = self.collate_fn([self.dataset[i] for i in indices])
    616             if self.pin_memory:
    617                 batch = pin_memory_batch(batch)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torch\utils\data\dataloader.py in <listcomp>(.0)
    613         if self.num_workers == 0:  # same-process loading
    614             indices = next(self.sample_iter)  # may raise StopIteration
--> 615             batch = self.collate_fn([self.dataset[i] for i in indices])
    616             if self.pin_memory:
    617                 batch = pin_memory_batch(batch)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torchvision\datasets\mnist.py in __getitem__(self, index)
     93 
     94         if self.transform is not None:
---> 95             img = self.transform(img)
     96 
     97         if self.target_transform is not None:

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torchvision\transforms\transforms.py in __call__(self, img)
     58     def __call__(self, img):
     59         for t in self.transforms:
---> 60             img = t(img)
     61         return img
     62 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torchvision\transforms\transforms.py in __call__(self, tensor)
    161             Tensor: Normalized Tensor image.
    162         """
--> 163         return F.normalize(tensor, self.mean, self.std, self.inplace)
    164 
    165     def __repr__(self):

~\AppData\Local\Continuum\anaconda3\lib\site-packages\torchvision\transforms\functional.py in normalize(tensor, mean, std, inplace)
    206     mean = torch.tensor(mean, dtype=torch.float32)
    207     std = torch.tensor(std, dtype=torch.float32)
--> 208     tensor.sub_(mean[:, None, None]).div_(std[:, None, None])
    209     return tensor
    210 

RuntimeError: output with shape [1, 28, 28] doesn't match the broadcast shape [3, 28, 28]

Is there a problem with my packages ? Even the solution doesn't work so I cannot compute the loss. 
Heeeelp

**Comment by ahlusar1989** [comment]
@temanava No there is no problem. You need to use the following transform:

```
[transform = transforms.Compose([
transforms.ToTensor(), transforms.Normalize([0.5], [0.5])])]
```

**Comment by ahlusar1989** [comment]
With the maintainers permission I would like to propose a PR.

**Comment by Ab-Jannadi** [comment]
Try this one:

dataiter = iter(trainloader)
images, labels = next(dataiter)

**Comment by Ab-Jannadi** [comment]
Try this one:

`dataiter = iter(trainloader)
images, labels = next(dataiter)`

**Comment by Ab-Jannadi** [comment]
Try this one:

`dataiter = iter(trainloader)

images, labels = next(dataiter)`

**Comment by Ab-Jannadi** [comment]
Try this one:

`dataiter = iter(trainloader)`
`images, labels = next(dataiter)`

**Comment by abhiojha8** [comment]
Closing this as it is fixed in PR #151
